topostats/processing.py

```diff
@@ -1150,6 +1150,18 @@ def process_scan(
         )
         topostats_object["height_profiles"] = height_profiles
 
+        if grainstats_df is None:
+            LOGGER.warning(f"[{topostats_object['filename']}] : Grainstats dataframe is empty, stopping processing.")
+            # Early return to stop further processing
+            return (
+                topostats_object["img_path"],
+                grainstats_df,
+                height_profiles,
+                {},  # empty image_stats dictionary
+                pd.DataFrame(),  # empty disordered_tracing_stats DataFrame
```

Comment:
We have the `topostats.utils.create_empty_dataframe()` function which creates empty dataframes with the expected columns (pulled from the dictionary `COLUMN_SETS` which is defined at the top of the module (it includes sets for both `disordered_tracing_statistics` and `mol_statistics`).

```suggestion
                create_empty_dataframe(column_set="disordered_tracing_statistics"),  # empty disordered_tracing_stats DataFrame
```

Can use similar for the empty `molstats_df`

(Function is already imported)